src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -280,16 +286,22 @@ float64_t CCrossValidation::evaluate_one_run()
 			/* set feature subset for training */
 			SGVector<index_t> inverse_subset_indices=
 					m_splitting_strategy->generate_subset_inverse(i);
-			m_features->add_subset(inverse_subset_indices);
+
+			CFeatures* features=CFeatures::create_shallow_copy(m_features);
+			SG_REF(features);
+			features->add_subset(inverse_subset_indices);
 			for (index_t p=0; p<m_features->get_num_preprocessors(); p++)
 			{
 				CPreprocessor* preprocessor = m_features->get_preprocessor(p);
-				preprocessor->init(m_features);
+				preprocessor->init(features);
```

Comment:
Actually, I think this is wrong anyways --- for now remove all the preprocessors from the xvalidation code, we can do that later